disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/storage/files/DefaultTemporaryFileProvider.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.storage.files;
+
+import java.io.File;
+import java.io.IOException;
+
+public class DefaultTemporaryFileProvider implements TemporaryFileProvider {
+  public static final TemporaryFileProvider INSTANCE = new DefaultTemporaryFileProvider();
```

Comment:
We prefer exposing singletons via public static methods rather than public static fields:
```
private static final TemporaryFileProvider INSTANCE = new DefaultTemporaryFileProvider();

public static TemporaryFileProvider getInstance() {
  return INSTANCE;
}
```

Reply:
Thanks, I've applied this change to all the cases where an INSTANCE field was directly accessed.